src/main/scala/io/altoo/akka/serialization/kryo/KryoSerializerBackend.scala

```diff
@@ -26,24 +26,29 @@ private[kryo] class KryoSerializerBackend(val kryo: Kryo, val bufferSize: Int, v
         log.error(e, "Could not serialize class with potentially circular references: {}", obj)
         throw new RuntimeException("Could not serialize class with potential circular references: " + obj)
     } finally {
-      releaseBuffer(buffer)
+      buffer.reset()
     }
   }
 
   // "fromBinary" deserializes the given array,
   // using the type hint (if any, see "includeManifest" above)
   // into the optionally provided classLoader.
   override def fromBinary(bytes: Array[Byte], clazz: Option[Class[_]]): AnyRef = {
-    if (includeManifest)
-      clazz match {
-        case Some(c) => kryo.readObject(getInput(bytes), c).asInstanceOf[AnyRef]
-        case _ => throw new RuntimeException("Object of unknown class cannot be deserialized")
-      }
-    else
-      kryo.readClassAndObject(getInput(bytes))
+    val buffer = getInput(bytes)
+    try {
+      if (includeManifest)
+        clazz match {
+          case Some(c) => kryo.readObject(buffer, c).asInstanceOf[AnyRef]
+          case _ => throw new RuntimeException("Object of unknown class cannot be deserialized")
+        }
+      else
+        kryo.readClassAndObject(buffer)
+    } finally {
+      buffer.close()
```

Comment:
Currently it is a no-op since we use an array as input instead of a stream, but implementation or usage may change in the future.